toucan_connectors/hubspot/hubspot_connector.py

```diff
@@ -102,37 +113,36 @@ def build_authorization_url(self, **kwargs):
     def _get_access_token(self):
         return self.__dict__['_oauth2_connector'].get_access_token()
 
-    def _handle_pagination(self, endpoint_info, query_params, headers):
-        url = endpoint_info['url']
-        legacy = endpoint_info['legacy']
+    def _handle_pagination(
+        self, endpoint_info: Dict[str, Any], data_name: str, query_params: Dict[str, str], headers
```

Comment:
i may be wrong, but the `data_name` you're calling in this method should not be the `dataset` ? based on the method `_retrieve_data` at line 155... 🤔